src/price/weightedMean.ts

```diff
@@ -0,0 +1,13 @@
+import assert from 'assert';
+
+export const weightedMean = (weightedValues: [number, number][]): number => {
+  const totalWeight = weightedValues.reduce(function (sum, weightedValue) {
+    assert(weightedValue[1] >= 0, 'Weight should not be negative');
```

Comment:
I would better return undefined and do not stop code from executing. we can filter out undefined and still be able to submit what we have.

Reply:
we can still handle exceptions if we want to process it no matter what. I would leave this for now in case if we feed the function with empty input, which will never happen.